src/reducers/author.js

```diff
@@ -31,9 +34,25 @@ export const author = (state = initialStates, action = {}) => {
         isFetching: true
       })
     case types.FETCH_AUTHOR_COLLECTION_SUCCESS:
-      let actionObjToAdd = _.omit(action, [ 'type', 'authorId' ])
-      actionObjToAdd.isFetching = false
-      return _.mergeWith({}, state, actionObjToAdd, concatIfIsArray)
+      const {
+        authorId,
+        response,
+        currentPage,
+        totalResults,
+        totalPages,
+        receivedAt } = action
+      const saveToState = {
+        authorId,
+        isFetching: false,
+        [authorId]: {
+          collectIndexList: _.get(response, 'result', []),
+          currentPage,
+          totalResults,
+          receivedAt,
+          isFinish: (currentPage - REQUEST_PAGE_START_FROM + 1 >= totalPages)
+        }
+      }
+      return _.mergeWith({}, state, saveToState, concatCollectIndexList)
```

Comment:
All you need is to merge the `collectIndexList`, so I think you can directly concat the collectIndexList and uniq it without using `mergeWith` function.